var/spack/repos/builtin/packages/hpcc/package.py

```diff
@@ -0,0 +1,208 @@
+                        "HPCC need fftw2 interface, "
+                        "here are brief notes how to make one:\n"
+                        "# make fftw2x_cdft interface to mkl\n"
+                        "cd $MKLROOT/interfaces/fftw2x_cdft\n"
+                        "make libintel64 PRECISION=MKL_DOUBLE "
+                        "interface=ilp64 MKLROOT=$MKLROOT\n"
+                        "\n"
+                        "# make FFTW C wrapper library\n"
+                        "cd $MKLROOT/interfaces/fftw2xc\n"
+                        "make libintel64 PRECISION=MKL_DOUBLE "
+                        "MKLROOT=$MKLROOT\n")
+                lin_alg_libs.append(libfftw2xc)
+                lin_alg_libs.append(libfftw2x_cdft)
+
+        # Linear Algebra library (BLAS or VSIPL)
+        self.config['@LAINC@'] = '-I{0}'.format(spec['blas'].prefix.include)
+        lin_alg_libs = lin_alg_libs + [
+            lib for lib in spec['blas'].libs if lib not in lin_alg_libs]
+
+        # pack all LA/FFT libraries
+        self.config['@LALIB@'] = ' '.join(lin_alg_libs)
+
+        # Compilers / linkers - Optimization flags
+        self.config['@CC@'] = '{0}'.format(spec['mpi'].mpicc)
+
+        # Compiler flags for CPU architecture optimizations
+        if spec.satisfies('%intel') and \
+                (spec.satisfies('^intel-parallel-studio+mpi') or
+                 spec.satisfies('^intel-mpi')):
+            # with intel-parallel-studio+mpi the '-march' arguments
+            # are not passed to icc
+            arch_opt = spec.target.optimization_flags(
+                spec.compiler.name, spec.compiler.version)
+            self.config['@CCFLAGS@'] = '-O3 {0}'.format(arch_opt)
+        self._write_make_arch(spec, prefix)
+
+    def build(self, spec, prefix):
+        make('arch={0}'.format(self.arch))
+
+    def check(self):
+        """Simple check that compiled binary is working:
+        launch with 4 MPI processes and check that test finished."""
+        # copy input
+        copy('_hpccinf.txt', 'hpccinf.txt')
+        # run test
+        run = Executable(
+            join_path(os.path.dirname(self.spec['mpi'].mpicc), 'mpirun'))
+        run('-np', '4', './hpcc')
+        # check output
+        hpccoutf = open('hpccoutf.txt', 'rt').read()
+        if not re.search("End of HPC Challenge tests", hpccoutf):
+            raise Exception("Test run was not successfull!")
+
+    def installcheck(self):
+        """Same as check but within prefix location"""
+        with working_dir(self.prefix.share.hpcc):
+            # run test
+            run = Executable(
+                join_path(os.path.dirname(self.spec['mpi'].mpicc), 'mpirun'))
+            run('-np', '4', self.prefix.bin.hpcc)
+            # check output
+            hpccoutf = open('hpccoutf.txt', 'rt').read()
+            if not re.search("End of HPC Challenge tests", hpccoutf):
+                raise Exception("Test run was not successfull!")
```

Comment:
I'm not sure if it's a good idea to run `mpi` tests during `install --test`, as some users may be installing this on a login node.

Reply:
I think to be able to run a test to ensure that application is properly compiled and working as expected is a very good idea. In spack running tests is a viable option, which can be turned on or turned off (default behavior). So only aware users will do that.

With upcoming zoo of ARM CPUs and given that some scientific codes are extremely sensitive to optimization flags/compilers and so on, I think the ability to run tests which spack have is a very important feature. I obviously like it. I have another code (Enzo, for astrophysical simulation), for which I already made an packaging within spack, but I can not make tests working within spack yet. That code do have some hard time with certain compilers and optimization flags.

Anyway that test is very tiny, it uses 4 mpi processes and few seconds to run at most, much less than compiling itself. So it would not bather more than regular compilation.

Reply:
In the long run, #15702 will add support for tests that can be run after the installation has finished. This will probably work much better for tests like this that require MPI and should be submitted to a queue. You can keep it as is for now, but once #15702 has been merged we should probably run this test not at build-time but at `spack test` time.

Reply:
Thanks for info